### PR TITLE
Enable authorization to Google Cloud Platform 

### PR DESCRIPTION
**Pull request by VJalili**
This PR enables authorization to Google Cloud Platform (GCP). Accordingly, it implements the following extensions: 

- implements means of configuring Galaxy with the instance-specific service account; 
- leverages cloudauthz v0.3.0 to obtain authorization to user's private resources on GCP;
- extends cloudauthz API and manager to support GCP.

_In subsequent PR(s), we will leverage these extensions to enable storing user data on Google cloud storage (GCS), and importing their data from GCS onto their Galaxy history._

### CODE REVIEW

**Comment by VJalili** [comment]
ping @afgane 

@nsoranzo how does dependencies change look to you?

File: lib/galaxy/dependencies/pipfiles/default/pinned-requirements.txt [hunk 1]
```diff
@@ -111,7 +111,7 @@ oslo.utils==3.41.0
 packaging==19.0
 paramiko==2.5.0
 parsley==1.3
-paste==3.0.8
+paste==3.0.8 ; python_version == '2.7'
```
Comment (nsoranzo) [inline lib/galaxy/dependencies/pipfiles/default/pinned-requirements.txt]: This is wrong, it was fixed in commit fed80f19f6340f01401390c1ef5850e1fd71153a, you need to re-run `make update-dependencies` now that you have merged the upstream `dev` branch.

File: lib/galaxy/dependencies/pipfiles/default/pinned-requirements.txt [hunk 2]
```diff
@@ -165,18 +165,18 @@ schema-salad==2.7.20181126142424
 shellescape==3.4.1
 simplejson==3.16.0
 six==1.11.0
-social-auth-core[openidconnect]==3.1.0+gx0
+social-auth-core[openidconnect]==3.1.0
```
Comment (nsoranzo) [inline lib/galaxy/dependencies/pipfiles/default/pinned-requirements.txt]: As above, this was fixed in commit fed80f19f6340f01401390c1ef5850e1fd71153a, need to re-run `make update-dependencies`.

File: lib/galaxy/dependencies/pipfiles/default/pinned-requirements.txt (hunk 1 above)
Comment (VJalili) [reply lib/galaxy/dependencies/pipfiles/default/pinned-requirements.txt]: @nsoranzo Thanks, should be fixed now.

File: lib/galaxy/dependencies/pipfiles/default/pinned-requirements.txt (hunk 2 above)
Comment (VJalili) [reply lib/galaxy/dependencies/pipfiles/default/pinned-requirements.txt]: @nsoranzo Thanks, should be fixed now.

File: lib/galaxy/authnz/managers.py [hunk 3]
```diff
@@ -304,10 +304,14 @@ def get_cloud_access_credentials(self, cloudauthz, sa_session, user_id, request=
...
                 cloudauthz.id, user_id))
-            return ca.authorize(cloudauthz.provider, config)
+            credentials = ca.authorize(cloudauthz.provider, config)
+            return credentials
         except CloudAuthzBaseException as e:
             log.info(e)
             raise exceptions.AuthenticationFailed(e)
+        except NotImplementedError as e:
+            log.info(e.message)
+            raise exceptions.RequestParameterInvalidException(e.message)
```
Comment (nsoranzo) [inline lib/galaxy/authnz/managers.py]: Exceptions don't have the `message` attribute any more under Python 3
```suggestion
            raise exceptions.RequestParameterInvalidException(e)
```

File: lib/galaxy/authnz/managers.py [hunk 4]
```diff
@@ -304,10 +304,14 @@ def get_cloud_access_credentials(self, cloudauthz, sa_session, user_id, request=
...
             log.info("Requesting credentials using CloudAuthz with config id `{}` on be half of user `{}`.".format(
                 cloudauthz.id, user_id))
-            return ca.authorize(cloudauthz.provider, config)
+            credentials = ca.authorize(cloudauthz.provider, config)
+            return credentials
         except CloudAuthzBaseException as e:
             log.info(e)
             raise exceptions.AuthenticationFailed(e)
+        except NotImplementedError as e:
+            log.info(e.message)
```
Comment (nsoranzo) [inline lib/galaxy/authnz/managers.py]:
```suggestion
            log.info(e)
```

File: lib/galaxy/authnz/managers.py (hunk 3 above)
Comment (VJalili) [reply lib/galaxy/authnz/managers.py]: Good to know, thanks!

**Pull request merged**